### PR TITLE
fix: prevent refocusing in options-selector and selection-list

### PR DESCRIPTION
**Pull request by dummy-1111**
### Fixed Issues

$ https://github.com/Expensify/App/issues/22812
PROPOSAL: https://github.com/Expensify/App/issues/22812#issuecomment-1634440613

### Tests
1. Open ND and log in with any account
2. Tap FAB and select `Send message`
3. Select multiple users by tapping `Add to group` button
4. Verify that the style of search input doesn't change
5. Close right modal and tap FAB and select `Request money`
6. Input any amount in manual tab and tap `Next`
7. Select multiple users by tapping `Split` button
8. Verify that the style of search input doesn't change
9. Go to settings -> workspace -> members
10. Check/uncheck members and verify that the style of search input doesn't change
11. Go to settings -> workspace -> members -> invite
12. Check/uncheck members and verify that the style of search input doesn't change

- [x] Verify that no errors appear in the JS console

### Offline tests
1. Open ND and log in with any account
2. Tap FAB and select `Send message`
3. Select multiple users by tapping `Add to group` button
4. Verify that the style of search input doesn't change
5. Close right modal and tap FAB and select `Request money`
6. Input any amount in manual tab and tap `Next`
7. Select multiple users by tapping `Split` button
8. Verify that the style of search input doesn't change
9. Go to settings -> workspace -> members
10. Check/uncheck members and verify that the style of search input doesn't change
11. Go to settings -> workspace -> members -> invite
12. Check/uncheck members and verify that the style of search input doesn't change

### QA Steps
**_Note_**: All the tests should be done on ios/android native, web, and desktop platforms. In mobile web platforms, search input gets blurred on selection by intention

1. Open ND and log in with any account
2. Tap FAB and select `Send message`
3. Select multiple users by tapping `Add to group` button
4. Verify that the style of search input doesn't change
5. Close right modal and tap FAB and select `Request money`
6. Input any amount in manual tab and tap `Next`
7. Select multiple users by tapping `Split` button
8. Verify that the style of search input doesn't change
9. Go to settings -> workspace -> members
10. Check/uncheck members and verify that the style of search input doesn't change
11. Go to settings -> workspace -> members -> invite
12. Check/uncheck members and verify that the style of search input doesn't change

- [x] Verify that no errors appear in the JS console

### PR Author Checklist

- [x] I linked the correct issue in the `### Fixed Issues` section above
- [x] I wrote clear testing steps that cover the changes made in this PR
    - [x] I added steps for local testing in the `Tests` section
    - [x] I added steps for the expected offline behavior in the `Offline steps` section
    - [x] I added steps for Staging and/or Production testing in the `QA steps` section
    - [x] I added steps to cover failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [x] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
    - [x] I tested this PR with a [High Traffic account](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#high-traffic-accounts) against the staging or production API to ensure there are no regressions (e.g. long loading states that impact usability).
- [x] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I ran the tests on **all platforms** & verified they passed on:
    - [x] Android / native
    - [x] Android / Chrome
    - [x] iOS / native
    - [x] iOS / Safari
    - [x] MacOS / Chrome / Safari
    - [x] MacOS / Desktop
- [x] I verified there are no console errors (if there's a console error not related to the PR, report it or open an issue for it to be fixed)
- [x] I followed proper code patterns (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [x] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`)
    - [x] I verified that the left part of a conditional rendering a React component is a boolean and NOT a string, e.g. `myBool && <MyComponent />`.
    - [x] I verified that comments were added to code that is not self explanatory
    - [x] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [x] I verified any copy / text shown in the product is localized by adding it to `src/languages/*` files and using the [translation method](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60)
      - [x] If any non-english text was added/modified, I verified the translation was requested/reviewed in #expensify-open-source and it was approved by an internal Expensify engineer. Link to Slack message:
    - [x] I verified all numbers, amounts, dates and phone numbers shown in the product are using the [localization methods](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60-L68)
    - [x] I verified any copy / text that was added to the app is grammatically correct in English. It adheres to proper capitalization guidelines (note: only the first word of header/labels should be capitalized), and is approved by marketing by adding the `Waiting for Copy` label for a copy review on the original GH to get the correct copy.
    - [x] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [x] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [x] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [x] I followed the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [x] I tested other components that can be impacted by my changes (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` are working as expected)
- [x] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [x] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [x] I verified that if a function's arguments changed that all usages have also been updated correctly
- [x] If a new component is created I verified that:
    - [x] A similar component doesn't exist in the codebase
    - [x] All props are defined accurately and each prop has a `/** comment above it */`
    - [x] The file is named correctly
    - [x] The component has a clear name that is non-ambiguous and the purpose of the component can be inferred from the name alone
    - [x] The only data being stored in the state is data necessary for rendering and nothing else
    - [x] If we are not using the full Onyx data that we loaded, I've added the proper selector in order to ensure the component only re-renders when the data it is using changes
    - [x] For Class Components, any internal methods passed to components event handlers are bound to `this` properly so there are no scoping issues (i.e. for `onClick={this.submit}` the method `this.submit` should be bound to `this` in the constructor)
    - [x] Any internal methods bound to `this` are necessary to be bound (i.e. avoid `this.submit = this.submit.bind(this);` if `this.submit` is never passed to a component event handler like `onClick`)
    - [x] All JSX used for rendering exists in the render method
    - [x] The component has the minimum amount of code necessary for its purpose, and it is broken down into smaller components in order to separate concerns and functions
- [x] If any new file was added I verified that:
    - [x] The file has a description of what it does and/or why is needed at the top of the file if the code is not self explanatory
- [x] If a new CSS style is added I verified that:
    - [x] A similar style doesn't already exist
    - [x] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/styles/StyleUtils.js) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(themeColors.componentBG)`)
- [x] If the PR modifies code that runs when editing or sending messages, I tested and verified there is no unexpected behavior for all supported markdown - URLs, single line code, code blocks, quotes, headings, bold, strikethrough, and italic.
- [x] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [x] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [x] If the PR modifies a component or page that can be accessed by a direct deeplink, I verified that the code functions as expected when the deeplink is used - from a logged in and logged out account.
- [x] If a new page is added, I verified it's using the `ScrollView` component to make it scrollable when more elements are added to the page.
- [x] If the `main` branch was merged into this PR after a review, I tested again and verified the outcome was still expected according to the `Test` steps.
- [x] I have checked off every checkbox in the PR author checklist, including those that don't apply to this PR.

### Screenshots/Videos
<details>
<summary>Web</summary>

https://github.com/Expensify/App/assets/126839717/13ee48c0-4628-4bb4-bc57-6dad6abb14da

</details>

<details>
<summary>Mobile Web - Chrome</summary>

https://github.com/Expensify/App/assets/126839717/1937a523-124e-4d5d-9db2-d46f088f95dd

</details>

<details>
<summary>Mobile Web - Safari</summary>

https://github.com/Expensify/App/assets/126839717/f52b1ce7-6607-41cd-9baf-0bd812b586ed

</details>

<details>
<summary>Desktop</summary>

https://github.com/Expensify/App/assets/126839717/52d68f04-6c49-487a-ba39-92fd24475ac8

</details>

<details>
<summary>iOS</summary>

https://github.com/Expensify/App/assets/126839717/fe88b954-a799-448d-aea7-54e7cc978d7c

</details>

<details>
<summary>Android</summary>

https://github.com/Expensify/App/assets/126839717/1397f1c1-9de0-4afa-8639-93a86d2b059b

</details>

### CODE REVIEW

File: src/components/OptionRow.js [hunk 1]
```diff
@@ -70,6 +70,9 @@ const propTypes = {
     /** Whether to remove the lateral padding and align the content with the margins */
     shouldDisableRowInnerPadding: PropTypes.bool,
 
+    /** Whether to take focus when selecting */
+    shouldTakeFocus: PropTypes.bool,
```
Comment (situchan) [inline src/components/OptionRow.js:74]: Can we make the name consistent? So use `shouldFocusOnSelectRow` and reverse conditions.

Comment (s-alves10) [reply src/components/OptionRow.js:74]: @situchan 

When we should refocus on selecting row for options-selector, the option row should not take focus. I thought this makes sense.
Do you still want to rename the props?

Comment (situchan) [reply src/components/OptionRow.js:74]: yeah, tricky naming. `shoudTakeFocus` is fine but no one knows what this does when read this code without any context.

Comment (situchan) [reply src/components/OptionRow.js:74]: Maybe `shouldPreventDefaultFocusOnClick` though it's long name?

Comment (s-alves10) [reply src/components/OptionRow.js:74]: I think it's fine

Comment (situchan) [reply src/components/OptionRow.js:74]: Let's do that for now. Btw Monil might address this again during final review.

Comment (s-alves10) [reply src/components/OptionRow.js]: @situchan 

Props name was changed.

**Comment by s-alves10** [comment]
@situchan 

Workspace invite page was updated. Videos were updated as well. Please take a look

**Comment by situchan** [comment]
@s-alves10 please correct QA Steps so that QA team doesn't get confused
- add invite page case
- mWeb behavior is different

**Comment by s-alves10** [comment]
@situchan 

I'm sorry I forgot to update that. Just updated

**Comment by situchan** [comment]
## Reviewer Checklist

- [x] I have verified the author checklist is complete (all boxes are checked off).
- [x] I verified the correct issue is linked in the `### Fixed Issues` section above
- [x] I verified testing steps are clear and they cover the changes made in this PR
    - [x] I verified the steps for local testing are in the `Tests` section
    - [x] I verified the steps for Staging and/or Production testing are in the `QA steps` section
    - [x] I verified the steps cover any possible failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [x] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
- [x] I checked that screenshots or videos are included for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I verified tests pass on **all platforms** & I tested again on:
    - [x] Android / native
    - [x] Android / Chrome
    - [x] iOS / native
    - [x] iOS / Safari
    - [x] MacOS / Chrome / Safari
    - [x] MacOS / Desktop
- [x] If there are any errors in the console that are unrelated to this PR, I either fixed them (preferred) or linked to where I reported them in Slack
- [x] I verified proper code patterns were followed (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [x] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`).
    - [x] I verified that the left part of a conditional rendering a React component is a boolean and NOT a string, e.g. `myBool && <MyComponent />`.
    - [x] I verified that comments were added to code that is not self explanatory
    - [x] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [x] I verified any copy / text shown in the product is localized by adding it to `src/languages/*` files and using the [translation method](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60)
    - [x] I verified all numbers, amounts, dates and phone numbers shown in the product are using the [localization methods](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60-L68)
    - [x] I verified any copy / text that was added to the app is grammatically correct in English. It adheres to proper capitalization guidelines (note: only the first word of header/labels should be capitalized), and is approved by marketing by adding the `Waiting for Copy` label for a copy review on the original GH to get the correct copy.
    - [x] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [x] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [x] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [x] I verified that this PR follows the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [x] I verified other components that can be impacted by these changes have been tested, and I retested again (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` have been tested & I retested again)
- [x] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [x] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [x] If a new component is created I verified that:
    - [x] A similar component doesn't exist in the codebase
    - [x] All props are defined accurately and each prop has a `/** comment above it */`
    - [x] The file is named correctly
    - [x] The component has a clear name that is non-ambiguous and the purpose of the component can be inferred from the name alone
    - [x] The only data being stored in the state is data necessary for rendering and nothing else
    - [x] For Class Components, any internal methods passed to components event handlers are bound to `this` properly so there are no scoping issues (i.e. for `onClick={this.submit}` the method `this.submit` should be bound to `this` in the constructor)
    - [x] Any internal methods bound to `this` are necessary to be bound (i.e. avoid `this.submit = this.submit.bind(this);` if `this.submit` is never passed to a component event handler like `onClick`)
    - [x] All JSX used for rendering exists in the render method
    - [x] The component has the minimum amount of code necessary for its purpose, and it is broken down into smaller components in order to separate concerns and functions
- [x] If any new file was added I verified that:
    - [x] The file has a description of what it does and/or why is needed at the top of the file if the code is not self explanatory
- [x] If a new CSS style is added I verified that:
    - [x] A similar style doesn't already exist
    - [x] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/styles/StyleUtils.js) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(themeColors.componentBG`)
- [x] If the PR modifies code that runs when editing or sending messages, I tested and verified there is no unexpected behavior for all supported markdown - URLs, single line code, code blocks, quotes, headings, bold, strikethrough, and italic.
- [x] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [x] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [x] If the PR modifies a component or page that can be accessed by a direct deeplink, I verified that the code functions as expected when the deeplink is used - from a logged in and logged out account.
- [x] If a new page is added, I verified it's using the `ScrollView` component to make it scrollable when more elements are added to the page.
- [x] If the `main` branch was merged into this PR after a review, I tested again and verified the outcome was still expected according to the `Test` steps.
- [x] I have checked off every checkbox in the PR reviewer checklist, including those that don't apply to this PR.

### Screenshots/Videos
<details>
<summary>Web</summary>


https://github.com/Expensify/App/assets/108292595/7b3ebaf8-71fd-4a26-b107-cef8852d6907



</details>

<details>
<summary>Mobile Web - Chrome</summary>


https://github.com/Expensify/App/assets/108292595/8762d1c1-68f8-4774-923d-861fcf8784af



</details>

<details>
<summary>Mobile Web - Safari</summary>


https://github.com/Expensify/App/assets/108292595/8e30f7ee-d261-46fa-b79d-1ea0d3adb0b6



</details>

<details>
<summary>Desktop</summary>


https://github.com/Expensify/App/assets/108292595/ee3551c0-029a-49ed-8e48-7bf963216efd



</details>

<details>
<summary>iOS</summary>


https://github.com/Expensify/App/assets/108292595/e88ff23c-e590-48ee-8c5f-3d60e3395c5c



</details>

<details>
<summary>Android</summary>



</details>

**Review by situchan — APPROVED** [review]
:shipit: 

@MonilBhavsar all yours

**Comment by MonilBhavsar** [comment]
Yes, reviewing now.

File: src/pages/workspace/WorkspaceInvitePage.js [hunk 2]
```diff
@@ -231,6 +232,7 @@ function WorkspaceInvitePage(props) {
                             onConfirm={inviteUser}
                             showScrollIndicator
                             showLoadingPlaceholder={!didScreenTransitionEnd || !OptionsListUtils.isPersonalDetailsReady(props.personalDetails)}
+                            shouldFocusOnSelectRow={!Browser.isMobile()}
```
Comment (MonilBhavsar) [inline src/pages/workspace/WorkspaceInvitePage.js:235]: There are two props - `shouldFocusOnSelectRow` and `shouldPreventDefaultFocusOnSelect`. Can we make it one to make it easy to understand?

Comment (situchan) [reply src/pages/workspace/WorkspaceInvitePage.js:235]: @MonilBhavsar Can you please check [this discussion](https://github.com/Expensify/App/pull/29084#discussion_r1350309238)?

File: src/components/OptionRow.js (hunk 1 above)
Comment (MonilBhavsar) [reply src/components/OptionRow.js]: Yes, I found `shouldFocusOnSelectRow` and `shouldPreventDefaultFocusOnSelect` different and confusing if one doesn't know the context. I think we should use only one. Name and context wise both looks good.

File: src/pages/workspace/WorkspaceInvitePage.js (hunk 2 above)
Comment (MonilBhavsar) [reply src/pages/workspace/WorkspaceInvitePage.js:235]: Thanks! Commented there

File: src/components/OptionRow.js (hunk 1 above)
Comment (situchan) [reply src/components/OptionRow.js]: Do you agree to use `shouldFocusOnSelectRow` everywhere?

Comment (s-alves10) [reply src/components/OptionRow.js]: If we still want to have the same props name, I suggest `shouldPreventDefaultFocusOnSelectRow`

Comment (MonilBhavsar) [reply src/components/OptionRow.js]: `shouldPreventDefaultFocusOnSelectRow` sounds good to me. 
I think also `shouldPreventDefaultFocusOnSelectItem` will work. 
But yes, using same prop through out the tree will be better IMO

Comment (s-alves10) [reply src/components/OptionRow.js]: Okay. Let me rename the props for BaseOptionSelector, BaseOptionsList, BaseSelectionList, BaseListItem, and OptionRow

cc @situchan

**Comment by s-alves10** [comment]
@MonilBhavsar 

Props name updated. Please check

**Review by MonilBhavsar — APPROVED** [review]
Thanks!

**Pull request merged**